xchange-lgo/src/main/java/info/bitrich/xchangestream/lgo/LgoStreamingService.java

```diff
@@ -0,0 +1,60 @@
+package info.bitrich.xchangestream.lgo;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import info.bitrich.xchangestream.lgo.dto.LgoSubscription;
+import info.bitrich.xchangestream.service.netty.JsonNettyStreamingService;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.handler.codec.http.DefaultHttpHeaders;
+import io.netty.handler.codec.http.websocketx.PingWebSocketFrame;
+import org.knowm.xchange.lgo.service.LgoSignatureService;
+
+import java.io.IOException;
+import java.time.Duration;
+
+import static info.bitrich.xchangestream.service.netty.StreamingObjectMapperHelper.getObjectMapper;
+
+public class LgoStreamingService extends JsonNettyStreamingService {
+
+    private final LgoSignatureService signatureService;
+    private final String apiUrl;
+
+    LgoStreamingService(LgoSignatureService signatureService, String apiUrl) {
+        super(apiUrl, Integer.MAX_VALUE, Duration.ofSeconds(10), Duration.ofSeconds(15), 1);
+        this.apiUrl = apiUrl;
+        this.signatureService = signatureService;
+    }
+
+    @Override
+    protected void handleIdle(ChannelHandlerContext ctx) {
```

Comment:
Heh, it's a blunt instrument, but probably effective enough!

Reply:
Indeed :) As a matter of fact, it serves exactly this purpose of sending PING messages everytime we don't have data out of the socket after a given time. 
Of course, without a PONG timeout, it's not perfect, but at least in case on connectivity issues, I noticed that netty does a proper job of  closing everything.